engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/GetAnnotationNode.java

```diff
@@ -39,16 +38,21 @@ Object doExecute(
       @Cached ThunkExecutorNode thunkExecutorNode,
       @Cached ExpectStringNode expectStringNode,
       @Cached TypeOfNode typeOfNode) {
-    String methodName = expectStringNode.execute(method);
-
     Object targetTypeResult = typeOfNode.execute(target);
     if (targetTypeResult instanceof DataflowError error) {
       return error;
     }
 
     if (targetTypeResult instanceof Type targetType) {
-      ModuleScope scope = targetType.getDefinitionScope();
-      Function methodFunction = scope.lookupMethodDefinition(targetType, methodName);
+      Function methodFunction;
+      String methodName = null;
+      if (method instanceof UnresolvedSymbol symbol) {
+        methodFunction = symbol.resolveFor(targetType);
+      } else {
+        methodName = expectStringNode.execute(method);
+        ModuleScope scope = targetType.getDefinitionScope();
+        methodFunction = scope.lookupMethodDefinition(targetType, methodName);
+       }
```

Comment:
Why do we still allow 'raw text' input here?

It seems that the API would be simpler if we had one input type - `UnresolvedSymbol`.

Reply:
I am used to keep compatibility at almost all cost. The cost of keeping previous behavior in this case is low, acceptable. The idea of dropping support for the `Text` argument hasn't even come to my mind.

Reply:
The cost is that we increase the complexity of this function - making it harder to understand, test  and maintain if we need to change it in the future.

It's an internal function so I'm not sure what is the benefit of that? Especially as the old behaviour is actually leading to wrongly resolved methods - so we should discourage using it. If we still allow to use it, someone later may forget why this change was introduced, think that using the `Text` API is easier to use - and use it introducing a bug that the extension methods will not be correctly resolved somewhere.

Reply:
Support for `Text` removed in [883b17d](https://github.com/enso-org/enso/pull/7115/commits/883b17db89691352fbae2b3a7e78725e6e572a25)